envoy/stats/store.h

```diff
@@ -27,10 +27,53 @@ class Sink;
 class SinkPredicates;
 
 /**
- * A store for all known counters, gauges, and timers.
+ * Store keeps track of all Scopes created in it, and the Scopes manage
+ * individual stats. Each stat is defined in a scope. There is a single root
+ * scope created in the Store, and more sub-scopes can be created.
+ *
```

Comment:
could you clarify a little bit here: 
could sub-scopes create a tree-like scope hierarchy? and anything special about their life-cycle management?

The store can do iteration over all-stats in its-owned scoped, including sub-scopes(children,grand-children scopes...)?

Reply:
Yes good point. I added some more text. PTAL.